### PR TITLE
use deployer version for all deployment requests

### PR DESCRIPTION
**Pull request by baixinsui**
**Fixed https://github.com/eclipse-xpanse/xpanse/issues/1943**

**Based on https://github.com/eclipse-xpanse/tofu-maker/pull/69 and https://github.com/eclipse-xpanse/terraform-boot/pull/149**

- **Register service templates and  request service deployment when `terraform-boot` and `tofu-maker` used:**
image

Register service templates successfully:
image

Deploy services successfully:
image

Modify services successfully:
image

Destroy services successfully：
image

- **Register service templates and  request service deployment when `terraform-local` and `tofu-local` used:**
image

Register service templates successfully:
image

Deploy services successfully:
image

Modify services successfully:
image

Destroy services successfully：
image

### CODE REVIEW

**Comment by baixinsui** [comment]
Waiting tests already

File: modules/orchestrator/src/main/java/org/eclipse/xpanse/modules/orchestrator/deployment/DeployResult.java [hunk 1]
```diff
@@ -54,6 +54,9 @@ public class DeployResult {
     @Schema(description = "The private properties of the service instance deployment.")
     private Map<String, String> privateProperties = new HashMap<>();
 
+    @Schema(description = "The deployer tool with version for executing deployment scripts.")
+    private String deployerWithVersion;
```
Comment (swaroopar) [inline modules/orchestrator/src/main/java/org/eclipse/xpanse/modules/orchestrator/deployment/DeployResult.java:58]: deployerVersionUsed

File: modules/orchestrator/src/main/java/org/eclipse/xpanse/modules/orchestrator/deployment/DeploymentScriptValidationResult.java [hunk 2]
```diff
@@ -17,5 +17,6 @@
 public class DeploymentScriptValidationResult {
 
     private boolean valid;
+    private String deployerWithVersion;
```
Comment (swaroopar) [inline modules/orchestrator/src/main/java/org/eclipse/xpanse/modules/orchestrator/deployment/DeploymentScriptValidationResult.java:20]: this must be 'deployerVersionUsed'

**Review by swaroopar — CHANGES_REQUESTED** [review]

File: modules/deployment/src/main/java/org/eclipse/xpanse/modules/deployment/deployers/deployertools/DeployerToolUtils.java [hunk 3]
```diff
@@ -310,58 +310,80 @@ private String findBestVersion(Set<String> allAvailableVersions, String required
...
         }
-        String actualVersion = getExactVersionOfExecutor(executorFile.getAbsolutePath(),
+        String versionNumber = getVersionNumberOfExecutor(executorFile.getAbsolutePath(),
                 versionCommandOutputPattern);
-        if (StringUtils.isBlank(actualVersion)) {
+        if (StringUtils.isBlank(versionNumber)) {
             return false;
         }
-        return isVersionSatisfied(actualVersion, requiredOperator, requiredNumber);
+        return isVersionSatisfied(versionNumber, requiredOperator, requiredNumber);
```
[the remark this answers is not in the document]
Comment (baixinsui) [reply modules/deployment/src/main/java/org/eclipse/xpanse/modules/deployment/deployers/deployertools/DeployerToolUtils.java:335]: This method will be called in method for find best version in installed executors and the method for installing best version into the installation path. It is useful for the first case, but unnecessary for the second case. I will create a new method for the second case without `isVersionSatisified`.

**Comment by Alice1319** [comment]
LGTM

**Review by swaroopar — APPROVED** [review]
LGTM

**Pull request merged**